### ISSUE TITLE
Update collection

### ISSUE THREAD

**Issue by aguilera51284** [roboncode/orango#47]
when i try to update document i have a error
```
 let result = await Model.findByIdAndUpdate(id, data);
```
`error: Unhandled Rejection at: Promise`

How does update work?
I have to create a new Model `new Model().update()` ?

**Comment by aguilera51284** [comment]
ok, I just saw the error, you can not update a single field, you need all of them. to be able to update, otherwise you can not. besides that it does not throw the error of the fields.

how can you differentiate from the put and patch method?

**Comment by aguilera51284** [comment]
is there any way to avoid some fields? i try to set {strict: false} in this [line
](https://github.com/roboncode/orango/blob/master/lib/Model.js#L60) but it still does not work

**Comment by roboncode** [comment]
`strict` if false by default but you should only need to pass the fields you need. Take this example...

```js
let UserSchema = orango.Schema(
  {
    provider: {
      type: String,
      required: 'create',
      allow: [ CONSTS.AUTH_PROVIDERS.EMAIL, CONSTS.AUTH_PROVIDERS.PHONE ]
    },
    identifier: {
      type: String, // email, facebook, twitter, etc
      required: 'create'
    },
    verified: Boolean,
    profile: {
      firstName: String,
      lastName: String
    },
    passwordHash: { type: String, required: 'create' },
    created: Date,
    lastSignin: Date
  },
  {
    strict: true,
    indexes: [
      {
        type: 'hash',
        fields: [ 'identifier' ]
      }
    ]
  }
)

orango.model('User', UserSchema)
```

And I have a record like so...

```json
{
    "_key": "163706",
    "_id": "users/163706",
    "_rev": "_X1HMD9y--_",
    "provider": "email",
    "identifier": "roboncode@gmail.com",
    "profile": {
      "firstName": "Robert",
      "lastName": "Taylor"
    },
    "created": "2018-12-02T16:04:10.026Z"
 }
```

To update...

```js
 let user = await User.findByIdAndUpdate('163706', {
   profile: { firstName: 'Rob' }
 })

console.log(user)
```

```json
{
   "_key": "163706",
   "_id": "users/163706",
   "_rev": "_X1HMD9y--_",
   "provider": "email",
   "identifier": "roboncode@gmail.com",
   "profile": {
      "firstName": "Rob",
      "lastName": "Taylor"
   },
   "created": "2018-12-02T16:04:10.026Z"
}
```

**Comment by aguilera51284** [comment]
You can check, if this works with Joi

**Comment by roboncode** [comment]
@aguilera51284 - Please see if this issue if resolved in by installing `orango@next`. Examples can be found for orango next in the `next` branch 

```js
npm i orango@next
```

**Comment by roboncode** [comment]
@aguilera51284 - Please see if this issue if resolved in by installing orango@next. Examples can be found for orango next in the next branch

npm i orango@next
The API has undergone many breaking changes so be sure to review examples - https://github.com/roboncode/orango/tree/next/examples

**Comment by roboncode** [comment]
Orango@next has been released as release v1.0.0-alpha-2.1
